prov/shm/src/smr_msg.c

```diff
@@ -315,7 +315,15 @@ static ssize_t smr_generic_sendmsg(struct smr_ep *ep, const struct iovec *iov,
 		use_ipc = ofi_hmem_is_ipc_enabled(iface) &&
 				smr_desc->flags & FI_HMEM_DEVICE_ONLY &&
 				!(op_flags & FI_INJECT);
+
+		if (FI_VERSION_GE(smr_prov.fi_version, FI_VERSION(1,19)) &&
```

Comment:
We should be able to drop the version check here. The ofi_mr is cast from the internal shm pointer so should reflect the current version as long as it's initialized which is done [here](https://github.com/ofiwg/libfabric/blob/main/prov/util/src/util_mr_map.c#L260)

Reply:
Addressed